packages/cli/src/workflows/workflow.service.ee.ts

```diff
@@ -235,4 +241,98 @@ export class EnterpriseWorkflowService {
 			);
 		});
 	}
+
+	async transferOne(user: User, workflowId: string, toProjectId: string) {
+		// 1. get workflow
+		const workflow = await this.sharedWorkflowRepository.findWorkflowForUser(workflowId, user, [
+			'workflow:delete',
+		]);
+		NotFoundError.isDefinedAndNotNull(
+			workflow,
+			`Could not find workflow with the id "${workflowId}". Make sure you have the permission to delete it.`,
+		);
```

Comment:
I find type guards easier to read than conditional logic and creating an error. 

But I don't feel strongly about it, so it's up for debate.